### PR TITLE
A-AddRentalTransaction Features Improvements

### PR DESCRIPTION
**Pull request by SemiColonKen**
Change duration data type to int. duration can only be an int between 1 to 365
Change startDate datat type to LocalDate. startDate must be of valid format and date that exists.

Moved exception check for existing and valid CarLicensePlate from TransactionList to TransactionParser

Change the parser input change to match other methods. From "add-tx /p /u /d /s" to "add-tx /c /u /d /s"

Resolved test conflicts and merge conflicts

### CODE REVIEW

**Comment by AaronZZ10** [comment]
try to fix the compiling error

File: src/main/java/parser/HelpParser.java [hunk 1]
```diff
@@ -13,10 +13,10 @@ public class HelpParser {
         commands.put("add-user", "Adds a new customer to the system.");
         commands.put("remove-user", "Removes a customer from the system.");
         commands.put("list-users", "Lists all customers.");
-        commands.put("add-car", "Adds a new car to the fleet.");
+        commands.put("add-car /n [CAR_MODEL] /c [CAR_ID] /p [PRICE]", "Adds a new car to the fleet.");
         commands.put("remove-car", "Removes a car from the fleet.");
         commands.put("list-cars", "Lists all rented-out cars.");
-        commands.put("add-tx /p [CAR_LICENSE_PLATE] /u [BORROWER_NAME] /d [DURATION] /s [START_DATE dd-MM-yyyy]",
+        commands.put("add-tx /c [CAR_ID] /u [CUSTOMER_NAME] /d [DURATION] /s [START_DATE]",
```
Comment (AaronZZ10) [inline src/main/java/parser/HelpParser.java:19]: why remove "dd-MM-yyyy"?

Comment (SemiColonKen) [reply src/main/java/parser/HelpParser.java:19]: I thought it might be confusing

Comment (SemiColonKen) [reply src/main/java/parser/HelpParser.java]: added back in in

**Review by rexkoh425 — APPROVED** [review]
LGTM

**Pull request merged**